docs/source/model_design/programming_vivarium/debugging/index.rst

```diff
@@ -0,0 +1,190 @@
+  https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#sections
+  And then add it to the list of decorators above.
+
+.. _debugging_rt:
+
+===============
+Basic Debugging
+===============
+
+.. contents::
+   :local:
+   :depth: 2
+
+Debugging is an art and science that we couldn't hope to cover on this 
+single page. Rather than attempt such foolishness, we we review a few simple 
+techniques in the context of a targeted example here. If you find the 
+bug to be outside the scope of this page, it's likely time to ask an 
+engineer for help.  
+
+Making Sense of the Stack Trace
+-------------------------------
+
+The stack trace is an output that the model provides to show what 
+error message was received and trace back the flow of code to what 
+caused the error. Often the immediate line of code that generated the error will be 
+significantly removed from the line of code that will need to be changed 
+to fix the error. 
+
+Let's work through an example to understand this. Say a line of code 
+we edit in the :code:`loader.py` file filters a dataset incorrectly 
+such that it is actually empty. Nothing in that immediate block of 
+code catches this error - instead the provided dataframe is just 
+stored as empty. 
+
+From there this same, empty dataset is used in another :code:`loader.py` 
+function to calculate a rate. So our erroreously empty dataframe will be 
+multiplied by a dataframe with the correct data in it. 
+When pandas tries to by this multiplication on the dataframes, 
+it will find that the lengths do not match when they should. Pandas has 
+functions that run through a series of checks and if one fails, throws an 
+error message. 
+
+So the bottom line of the stack trace will likely say an error was thrown 
+in a pandas error message function that the lengths are not equal. From there, 
+the stack trace displays the code that lead here - for example the pandas 
+checking function that was run, then the pandas function that was called 
+(the multiplication), then the secondary 
+:code:`loader.py` function that tried to multiply with the empty dataframe. And still, 
+none of these will actually reveal the issue with filtering the dataset 
+we started with! This can be further complicated if the :code:`loader.py` function 
+using some functions for other packages, like :code:`vivarium_research` or :code:`vivarium_public_health`. 
+
+As you can see from this example, the stack trace can be both helpful and unhelpful. 
+In general, look through the files to see if you can get back to something 
+you wrote or edited - like the :code:`loader.py` file. If so, this is a good 
+place to start looking for issues, and the stack trace will provide the line 
+number of the code that caused the error. 
+
+Otherwise, see the next section for how to use the python debugger. 
+
+Navigating the Python Debugger
+------------------------------
+
+In all of the commands used to run engineering commands, the 
```

Comment:
What do you mean by "engineering commands"? Can you give one or more examples of a command you might be running where you can use the `--pdb` flag?